### PR TITLE
doc/start: make OSD and MDS structures parallel

### PR DESCRIPTION
**Pull request by zdover23**
This PR makes the "Ceph OSDs" and "MDSs" bullet points
parallel by naming "object storage daemon" before referring
to the (admittedly more common and colloquial, but surely
unknown to people who genuinely require a document called
'Intro') acronym "OSD".

Signed-off-by: Zac Dover <zac.dover@gmail.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/start/intro.rst [hunk 1]
```diff
@@ -34,7 +34,7 @@ required when running Ceph File System clients.
   `REST API`_.  At least two managers are normally required for high
   availability.
 
-- **Ceph OSDs**: A :term:`Ceph OSD` (object storage daemon,
+- **Ceph OSDs**: An object storage daemon (:term:`Ceph OSD`,
```
Comment (anthonyeleven) [inline doc/start/intro.rst:37]: Suggest:

**Ceph OSDs**: An Object Storage Daemon (:term:`Ceph OSD`,``ceph-osd``) stores data and handles replication, recovery, and rebalancing.  OSDs also provide some monitoring information to Ceph
...

I like to capitalize the letters that follow in the acronym to highlight the correspondence.

Comment (zdover23) [reply doc/start/intro.rst]: I've added those capitalized letters. Back to you for review.

**Comment by anthonyeleven** [comment]
Indeed, this is one of the places where Ceph overloads terms.  Not unreasonably, many people think D stands for Disk or Drive.

**Review by anthonyeleven — APPROVED** [review]

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/46656 - Pacific backport
https://github.com/ceph/ceph/pull/46655 - Quincy backport

**Pull request merged**